agent/lib/kontena/workers/service_pod_manager.rb

```diff
@@ -0,0 +1,124 @@
+require_relative 'service_pod_worker'
+require_relative '../models/service_pod'
+
+module Kontena::Workers
+  class ServicePodManager
+    include Celluloid
+    include Celluloid::Notifications
+    include Kontena::Logging
+    include Kontena::Helpers::RpcHelper
+    include Kontena::Helpers::WaitHelper
+
+    attr_reader :workers, :node
+
+    trap_exit :on_worker_exit
+    finalizer :finalize
+
+    def initialize(autostart = true)
+      @workers = {}
+      subscribe('agent:node_info', :on_node_info)
+      subscribe('service_pod:update', :on_update_notify)
+      async.start if autostart
+    end
+
+    def start
+      wait!(interval: 0.1, message: 'waiting for node info') { self.node }
```

Comment:
We have a problem here.. if actor crashes then node information is lost and we are not getting a new notification. Actor is basically in a restart loop after that. /cc @SpComb